phases/codesign.go

```
@@ -83,43 +93,140 @@ func AutoCodesign(bitriseYML bitriseModels.BitriseDataModel, searchDir string) (
+				Items: []string{answerYes, answerNo},
+				Templates: &promptui.SelectTemplates{
+					Label:    fmt.Sprintf("%s {{.}} ", promptui.IconInitial),
+					Selected: "Upload Android keystore file: {{ . | green }}",
+				},
+			}
+			_, uploadAndroid, err := prmpt.Run()
+			if err != nil {
+				return CodesignResult{}, fmt.Errorf("scan user input: %s", err)
+			}
+
+			if uploadAndroid == answerNo {
+				break
+			}
+
+			result.Android, err = getAndroidKeystoreSettings()
+			if err != nil {
+				log.Errorf("%s", err)
+				continue
+			}
+
+			break
+		}
+	}
+
+	return result, nil
+}
+
+func getAndroidKeystoreSettings() (CodesignResultAndroid, error) {
+	prompt := promptui.Prompt{
+		Label: "Enter keystore path",
+		Templates: &promptui.PromptTemplates{
+			Success: "Keystore path: {{ . | green }}",
+		},
+	}
+
+	var absKeystorePath string
+	{
+		keystorePath, err := prompt.Run()
 		if err != nil {
-			return CodesignResult{}, err
+			return CodesignResultAndroid{}, fmt.Errorf("scan user input: %s", err)
 		}
 
-		if uploadAndroid {
-			(&option{
-				title: "Enter key store path",
-				action: func(answer string) *option {
-					result.Android.KeystorePath = answer
-					return nil
-				}}).run()
-
-			(&option{
-				title:  "Enter key store password",
-				secret: true,
-				action: func(answer string) *option {
-					result.Android.Password = answer
-					return nil
-				}}).run()
-
-			(&option{
-				title: "Enter key alias",
-				action: func(answer string) *option {
-					result.Android.Alias = answer
-					return nil
-				}}).run()
-
-			(&option{
-				title:  "Enter key password",
-				secret: true,
-				action: func(answer string) *option {
-					result.Android.KeyPassword = answer
-					return nil
-				}}).run()
+		absKeystorePath, err = pathutil.AbsPath(keystorePath)
+		if err != nil {
+			return CodesignResultAndroid{}, fmt.Errorf("failed to get absolute keystore path, error: %s", err)
+		}
+		if _, err := os.Stat(absKeystorePath); os.IsNotExist(err) {
```

these checks should be a part of the validation checks

---

If the path is wrong, I think it is better to fail early. The keystore validation would also fail, but only after reading in 3 additional paramters.